pkg/api/namespace/formatter.go

```diff
@@ -0,0 +1,166 @@
+		} else if !ok {
+			return apierror.NewAPIError(validation.PermissionDenied, "User does not have permission to update resource quota")
+		}
+
+		var updateResourceQuotaInput UpdateResourceQuotaInput
+		if err := json.NewDecoder(req.Body).Decode(&updateResourceQuotaInput); err != nil {
+			return apierror.NewAPIError(validation.InvalidBodyContent, fmt.Sprintf("Failed to decode request body: %v ", err))
+		}
+		return h.updateResourceQuota(namespace, updateResourceQuotaInput)
+	case deleteResourceQuotaAction:
+		if ok, err := apiutil.CanUpdateResourceQuota(h.clientSet, namespace, user.GetName()); err != nil {
+			return apierror.NewAPIError(validation.ServerError, fmt.Sprintf("Failed to check permission: %v", err))
+		} else if !ok {
+			return apierror.NewAPIError(validation.PermissionDenied, "User does not have permission to update resource quota")
+		}
+		return h.deleteResourceQuota(namespace)
+	default:
+		return apierror.NewAPIError(validation.InvalidAction, "Unsupported action")
+	}
+}
+
+func (h *Handler) updateResourceQuota(namespace string, updateResourceQuotaInput UpdateResourceQuotaInput) error {
+	totalSnapshotSizeQuotaQuantity, err := resource.ParseQuantity(updateResourceQuotaInput.TotalSnapshotSizeQuota)
+	if err != nil {
+		return apierror.NewAPIError(validation.InvalidBodyContent, fmt.Sprintf("Failed to parse totalSnapshotSizeQuota: %v", err))
+	}
+	totalSnapshotSizeQuotaInt64, ok := totalSnapshotSizeQuotaQuantity.AsInt64()
+	if !ok {
+		return apierror.NewAPIError(validation.InvalidBodyContent, "Failed to parse totalSnapshotSizeQuota as int64")
+	}
+
+	if namespace == "" {
+		namespace = metav1.NamespaceDefault
+	}
+	resourceQuota, err := h.resourceQuotaClient.Get(namespace, util.DefaultResourceQuotaName, metav1.GetOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			resourceQuota = &harvesterv1.ResourceQuota{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: namespace,
+					Name:      util.DefaultResourceQuotaName,
+				},
+				Spec: harvesterv1.ResourceQuotaSpec{
+					SnapshotLimit: harvesterv1.SnapshotLimit{
+						NamespaceTotalSnapshotSizeQuota: totalSnapshotSizeQuotaInt64,
+					},
+				},
+			}
+			_, err = h.resourceQuotaClient.Create(resourceQuota)
+			return err
+		}
+		return err
+	}
+
+	resourceQuotaCopy := resourceQuota.DeepCopy()
+	resourceQuotaCopy.Spec.SnapshotLimit.NamespaceTotalSnapshotSizeQuota = totalSnapshotSizeQuotaInt64
+	if !reflect.DeepEqual(resourceQuota, resourceQuotaCopy) {
+		_, err = h.resourceQuotaClient.Update(resourceQuotaCopy)
+		return err
+	}
+	return nil
+}
+
+func (h *Handler) deleteResourceQuota(namespace string) error {
+	resourceQuota, err := h.resourceQuotaClient.Get(namespace, util.DefaultResourceQuotaName, metav1.GetOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil
+		}
+		return err
+	}
+
+	resourceQuotaCopy := resourceQuota.DeepCopy()
+	resourceQuotaCopy.Spec.SnapshotLimit.NamespaceTotalSnapshotSizeQuota = 0
```

Comment:
Wondering if we should remove the VM snapshot quota too.

Reply:
This is API for namespace, so we only remove namespace quota here. For VM, we have another API.